### PR TITLE
mp3val: fix patch

### PR DESCRIPTION
**Pull request by iMichka**
patch does not apply on Ventura

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
Don't know what is going with this patch. I looked for special characters, carriage returns, etc ... I can't get it to apply. I redid the patch myself and it's still not working.
I went for the inreplace strategy which seems to be working.

**Comment by carlocab** [comment]
Not really sure how that happened but it looks like @BrewTestBot messed with the patch somehow? https://github.com/Homebrew/homebrew-core/commit/07738c16c816fbc7d613dc5ea535122c852ca037

**Comment by iMichka** [comment]
Yes and I made multiple attempts to remake the patch myself, and each time I copy paste it something prevents it to be applied. 

This formula does not have a high download count and I don’t want to spend more time investigating

File: Formula/mp3val.rb [hunk 1]
```diff
@@ -18,12 +18,13 @@ class Mp3val < Formula
...
-  # Remove with next release.
-  patch :DATA
-
   def install
+    # Apply this upstream commit to fix build on Linux:
+    # https://sourceforge.net/p/mp3val/subversion/95/
+    # Remove with next release.
+    inreplace "crossapi.cpp",
+              "od=open(szNewName,O_WRONLY|O_CREAT|O_TRUNC);",
+              "od=open(szNewName,O_WRONLY|O_CREAT|O_TRUNC, S_IRUSR|S_IWUSR);"
```
Comment (chenrui333) [inline Formula/mp3val.rb:27]: Is this the same patch as before?

Also, this is patch only for linux build?

Comment (iMichka) [reply Formula/mp3val.rb:27]: Yes, I am replacing the same line as in the initial patch.
The initial patch was also applied on macOS and seems to do no harm (note that upstream did not make the change Linux specific)

**Review by chenrui333 — APPROVED** [review]

File: Formula/mp3val.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/mp3val.rb:27]: I see, 👍

**Pull request merged**